bottlerocket-settings-models/settings-extensions/container-runtime-plugins/src/de.rs

```diff
@@ -0,0 +1,209 @@
+use byte_unit::Byte;
+use serde::{self, Deserialize, Deserializer};
+
+/// This function deserializes a size string (e.g., "64mb") or integer into an i64 value.
+///
+/// Valid formats:
+/// - Integer values (e.g., "1024" or 1024)
+/// - Size strings with units (e.g., "64mb", "1.5gb", ".5gb")
+///
+/// Invalid formats:
+/// - Negative values other than -1
+/// - Strings with invalid formats
+/// - Strings with whitespace
+pub fn deserialize_chunk_size<'de, D>(deserializer: D) -> Result<Option<i64>, D::Error>
```

Comment:
We want the optionality to be applied to the struct field rather than to this value:
```suggestion
pub fn deserialize_chunk_size<'de, D>(deserializer: D) -> Result<i64, D::Error>
```